ripple/src/main/webapp/trie.js

```diff
@@ -0,0 +1,103 @@
+/* @source https://gist.github.com/tpae/72e1c54471e88b689f85ad2b3940a8f0 */
+
+/* Create TrieNode class. Used in Trie implementation */
+var Trie = function () {
+  var that = Object.create(Trie.prototype);
+  that.children = {}; // mapping: next character -> child nodes
+  that.isWord = false;
+  
+  /* Insert one word. */
+  that.insertWord = function (word) {
+    var current_node = that;
+    for (var i = 0; i < word.length; i++) {
+      var c = word[i]
```

Comment:
nit: I would choose a better variable name than `c` since you use it below a fair amount and its not always obvious what `c` is

Reply:
fixed!